redefinition of 'struct unity::util::internal::GlibDeleter<char>'

Now fails to build on Arch, bionic branch:

```
In file included from /home/vanyasem/Projects/AUR/_Unity8/_Unity8-Arch/unity-api-git/src/unity-api/test/gtest/unity/util/GioMemory/GioMemory_test.cpp:20:
/home/vanyasem/Projects/AUR/_Unity8/_Unity8-Arch/unity-api-git/src/unity-api/include/unity/util/GlibMemory.h:107:19: error: redefinition of 'struct unity::util::internal::GlibDeleter<char>'
 template<> struct GlibDeleter<TypeName> \
                   ^~~~~~~~~~~~~~~~~~~~~
/home/vanyasem/Projects/AUR/_Unity8/_Unity8-Arch/unity-api-git/src/unity-api/include/unity/util/GlibMemory.h:213:1: note: in expansion of macro 'UNITY_UTIL_DEFINE_GLIB_SMART_POINTERS'
 UNITY_UTIL_DEFINE_GLIB_SMART_POINTERS(gchar, g_free)
 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/vanyasem/Projects/AUR/_Unity8/_Unity8-Arch/unity-api-git/src/unity-api/include/unity/util/GlibMemory.h:107:19: note: previous definition of 'struct unity::util::internal::GlibDeleter<char>'
 template<> struct GlibDeleter<TypeName> \
                   ^~~~~~~~~~~~~~~~~~~~~
/home/vanyasem/Projects/AUR/_Unity8/_Unity8-Arch/unity-api-git/src/unity-api/include/unity/util/GlibMemory.h:192:55: note: in expansion of macro 'UNITY_UTIL_DEFINE_GLIB_SMART_POINTERS'
 #define G_DEFINE_AUTOPTR_CLEANUP_FUNC(TypeName, func) UNITY_UTIL_DEFINE_GLIB_SMART_POINTERS(TypeName, func)
                                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/glib-2.0/glib/glib-autocleanups.h:90:1: note: in expansion of macro 'G_DEFINE_AUTOPTR_CLEANUP_FUNC'
 G_DEFINE_AUTOPTR_CLEANUP_FUNC (GRefString, g_ref_string_release)
```

--- a/include/unity/util/GlibMemory.h
+++ b/include/unity/util/GlibMemory.h
@@ -210,7 +210,10 @@ inline GSourceManager g_source_manager(guint id)
 /**
  * Manually add extra definitions for gchar* and gchar**
  */
-#if !GLIB_CHECK_VERSION(2, 57, 2)
+#if GLIB_CHECK_VERSION(2, 57, 2)
+typedef GRefStringSPtr gcharSPtr;
+typedef GRefStringUPtr gcharUPtr;
+#else
 UNITY_UTIL_DEFINE_GLIB_SMART_POINTERS(gchar, g_free)
 #endif
 typedef gchar* gcharv;

--- a/include/unity/util/GlibMemory.h
+++ b/include/unity/util/GlibMemory.h
@@ -210,7 +210,9 @@ inline GSourceManager g_source_manager(guint id)
 /**
  * Manually add extra definitions for gchar* and gchar**
  */
+#if !GLIB_CHECK_VERSION(2, 57, 2)
 UNITY_UTIL_DEFINE_GLIB_SMART_POINTERS(gchar, g_free)
+#endif
 typedef gchar* gcharv;
 UNITY_UTIL_DEFINE_GLIB_SMART_POINTERS(gcharv, g_strfreev)
 

